src/lwc/geFormRenderer/geFormRenderer.js

```diff
@@ -883,9 +884,39 @@ export default class GeFormRenderer extends NavigationMixin(LightningElement) {
             }
         }
 
-        // Set DataImport__c.Additional_Object_JSON__c if we've retrieved values from form widgets.
+        // resolve widget data to wherever it needs to go, a field on the di object or elsewhere.
         if (widgetValues) {
-            diRecord[ADDITIONAL_OBJECT_JSON__C] = JSON.stringify(widgetValues);
+            let diFieldPayloads = [];
+
+            widgetValues.forEach(widgetValue => {
+                if(widgetValue.type === WIDGET_TYPE_DI_FIELD_VALUE && !isUndefined(widgetValue.payload)) {
+                    diFieldPayloads.push(widgetValue.payload);
+                }
+            });
+
+            // diFieldValues is an array of objects where the key is the field name
+            // and the value is the value to be stored
+            let diFieldValuesArray;
+            try {
+                diFieldValuesArray = await Promise.all(diFieldPayloads);
+            } catch (ex) {
+                // ex will be whatever was sent to the reject() function in the original promise
+                // TODO: handle payment error here: handlePaymentError(ex);
```

Comment:
I'm this hang on save when I have a card error (like expiration date in the past) - Is that related to this not being implemented? Console error shows "Uncaught (in promise) TypeError: Cannot read property 'forEach' of undefined
    at ie.buildDataImportRecord (geFormRenderer.js:4)"